samples/nrf9160/multicell_location/src/main.c

```diff
@@ -0,0 +1,235 @@
+		k_sem_give(&lte_connected);
+		break;
+	case LTE_LC_EVT_PSM_UPDATE:
+		LOG_DBG("PSM parameter update: TAU: %d, Active time: %d",
+			evt->psm_cfg.tau, evt->psm_cfg.active_time);
+		break;
+	case LTE_LC_EVT_EDRX_UPDATE: {
+		char log_buf[60];
+		ssize_t len;
+
+		len = snprintf(log_buf, sizeof(log_buf),
+			       "eDRX parameter update: eDRX: %f, PTW: %f",
+			       evt->edrx_cfg.edrx, evt->edrx_cfg.ptw);
+		if (len > 0) {
+			LOG_DBG("%s", log_strdup(log_buf));
+		}
+		break;
+	}
+	case LTE_LC_EVT_RRC_UPDATE:
+		LOG_DBG("RRC mode: %s",
+			evt->rrc_mode == LTE_LC_RRC_MODE_CONNECTED ?
+			"Connected" : "Idle");
+		break;
+	case LTE_LC_EVT_CELL_UPDATE:
+		LOG_DBG("LTE cell changed: Cell ID: %d, Tracking area: %d",
+			evt->cell.id, evt->cell.tac);
+		break;
+	case LTE_LC_EVT_LTE_MODE_UPDATE:
+		LOG_INF("Active LTE mode changed: %s",
+			evt->lte_mode == LTE_LC_LTE_MODE_NONE ? "None" :
+			evt->lte_mode == LTE_LC_LTE_MODE_LTEM ? "LTE-M" :
+			evt->lte_mode == LTE_LC_LTE_MODE_NBIOT ? "NB-IoT" :
+			"Unknown");
+		break;
+	case LTE_LC_EVT_NEIGHBOR_CELL_MEAS:
+		LOG_INF("Neighbor cell measurements received");
+
+		/* Copy current and neighbor cell information. */
+		memcpy(&cell_data, &evt->cells_info, sizeof(struct lte_lc_cells_info));
+		memcpy(neighbor_cells, evt->cells_info.neighbor_cells,
+			sizeof(struct lte_lc_ncell) * cell_data.ncells_count);
+		cell_data.neighbor_cells = neighbor_cells;
+
+		k_sem_give(&cell_data_ready);
+		break;
+	default:
+		break;
+	}
+}
+
+static int lte_connect(void)
+{
+	if (IS_ENABLED(CONFIG_LTE_AUTO_INIT_AND_CONNECT)) {
+		/* Do nothing, modem is already configured and LTE connected. */
+		return 0;
+	} else {
+		int err;
+
+		err = lte_lc_psm_req(true);
+		if (err) {
+			LOG_ERR("Failed to request PSM, error: %d", err);
+		}
+
+		err = lte_lc_edrx_req(true);
+		if (err) {
+			LOG_ERR("Failed to request eDRX, error: %d", err);
+		}
+
+		err = lte_lc_init_and_connect_async(lte_handler);
+		if (err) {
+			LOG_ERR("Modem could not be configured, error: %d",
+				err);
+			return err;
+		}
+
+		/* Check LTE events of type LTE_LC_EVT_NW_REG_STATUS in
+		 * lte_handler() to determine when the LTE link is up.
+		 */
+
+		return 0;
+	}
+}
+
+static void button_handler(uint32_t button_states, uint32_t has_changed)
+{
+	if (has_changed & button_states & DK_BTN1_MSK) {
+		int err = lte_lc_neighbor_cell_measurement();
+
+		if (err) {
+			LOG_ERR("Failed to initiate neighbor cell measurements");
+		}
+	}
+}
+
+static void periodic_search_work_fn(struct k_work *work)
+{
+	int err = lte_lc_neighbor_cell_measurement();
+
+	if (err) {
+		LOG_ERR("Failed to initiate neighbor cell measurements");
+	}
+
+	k_work_reschedule(k_work_delayable_from_work(work),
+		K_SECONDS(CONFIG_MULTICELL_LOCATION_SAMPLE_PERIODIC_SEARCH_INTERVAL));
+}
+
+static void print_cell_data(void)
+{
+	if (cell_data.current_cell.id == 0) {
+		LOG_WRN("No cells were found");
+		return;
+	}
+
+	printk("Current cell:\n");
```

Comment:
Why mixing up logger and printk?

Reply:
Waited to convert to upmerge is in and logger now handles floats too, changed now